### PR TITLE
Improve ConvND error message

### PR DESCRIPTION
**Pull request by marload**
Personally, I prefer to simplify it in one line rather than 'full if else syntax'.

Thank you for the efforts of DeepMind and the contributors to this wonderful project.

### CODE REVIEW

**Comment by marload** [comment]
An error was found in the a3c7b47 commit and the error was resolved in the 44cc50c commit.

File: haiku/_src/stateful.py [hunk 1]
```diff
@@ -122,12 +122,8 @@ def grad(fun, argnums=0, has_aux=False, holomorphic=False):
...
 
   def grad_fn(*args, **kwargs):
-    value, grads = value_and_grad_fun(*args, **kwargs)
-    if has_aux:
-      value, aux = value
-      return grads, aux
-    else:
-      return grads
+    (_, aux), grads = value_and_grad_fun(*args, **kwargs)
+    return (grads, aux if hax_aux else grads)
```
Comment (ibab) [inline haiku/_src/stateful.py]: Won't this return `(grads, grads)` when `hax_aux=False`? That would be different from what it did before.

Comment (marload) [reply haiku/_src/stateful.py]: You're right. The problem has been resolved and reflected. 44cc50

File: haiku/_src/stateful.py [hunk 2]
```diff
@@ -122,12 +122,8 @@ def grad(fun, argnums=0, has_aux=False, holomorphic=False):
                                       holomorphic=holomorphic)
 
   def grad_fn(*args, **kwargs):
-    value, grads = value_and_grad_fun(*args, **kwargs)
-    if has_aux:
-      value, aux = value
-      return grads, aux
-    else:
-      return grads
+    (_, aux), grads = value_and_grad_fun(*args, **kwargs)
```
Comment (trevorcai) [inline haiku/_src/stateful.py]: I don't think this is valid in the has_aux=False case.
Does the test suite pass?

File: haiku/_src/stateful.py [hunk 3]
```diff
@@ -190,10 +186,7 @@ def wrapper(*args, **kwargs):
     kwargs["hk_state"] = internal_state()
     (value, (aux, hk_state)), grads = grad_fun(*args, **kwargs)
     update_internal_state(hk_state)
-    if has_aux:
-      return (value, aux), grads
-    else:
-      return value, grads
+    return (value, aux) if has_aux else value, grads
```
Comment (trevorcai) [inline haiku/_src/stateful.py]: This is hard for me to parse - I initially misread it as returning `(value, aux)` else `(value, grads)`.
Acknowledging that the previous version is much more verbose, I'd rather have verbose code than hard-to-parse code. Consider refactoring again or reverting this chunk of the change.

**Comment by marload** [comment]
@trevorcai 
Thank you! I thought about your two opinions. I think you are right in both opinions. So I changed the two changed codes to the original code. Thanks to you, I learned a lot!

**Comment by marload** [comment]
@trevorcai 
I clicked detail to see why feedback/copybara failed, but there was no response. How do I fix this?

**Comment by trevorcai** [comment]
resolved internally, the pr will land when tom approves it internally.

**Pull request merged**